### PR TITLE
Update OR policies and env sampler workflow

### PR DESCRIPTION
**Pull request by Jinyu-W**
…use List[state] as input

# Description

- Move OR state shaper to an independent file
- Update OR policies to use List[state] as input

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/rl/rollout/env_sampler.py [hunk 1]
```diff
@@ -283,8 +283,6 @@ def _get_global_and_agent_state(
             policy = self._policy_dict[policy_name]
             if isinstance(policy, RLPolicy) and not isinstance(state, np.ndarray):
                 raise ValueError(f"Agent {agent_name} uses a RLPolicy but its state is not a np.ndarray.")
-            if not isinstance(policy, RLPolicy) and not isinstance(state, list):
```
Comment (lihuoran) [inline maro/rl/rollout/env_sampler.py:286]: Why remove this?

File: examples/supply_chain/rl/env_sampler.py [hunk 2]
```diff
@@ -143,75 +144,80 @@ def __init__(
...
         entity = self._entity_dict[entity_id]
 
         if isinstance(self._policy_dict[self._agent2policy[entity_id]], RLPolicy):
-            return self.get_rl_policy_state(entity_id, entity)
-        else:
+            return self.get_rl_policy_state(entity_id)
+        elif isinstance(self._policy_dict[self._agent2policy[entity_id]], ConsumerBasePolicy):
             return self.get_or_policy_state(entity)
+        else:
+            return []
```
Comment (lihuoran) [inline examples/supply_chain/rl/env_sampler.py:220]: Should not return a list.

File: examples/supply_chain/rl/algorithms/rule_based.py [hunk 3]
```diff
@@ -1,127 +1,70 @@
...
-
-def get_element(np_state: np.ndarray, key: str) -> np.ndarray:
-    offsets = np_state[0][-len(OR_STATE_OFFSET_INDEX):].astype(np.uint)
-    idx = OR_STATE_OFFSET_INDEX[key]
-    prev_idx = int(offsets[idx - 1]) if idx > 0 else 0
-    return np_state[:, prev_idx: offsets[idx]].squeeze()
 
+VLT_BUFFER_DAYS = workflow_settings["or_policy_vlt_buffer_days"]
 
 class DummyPolicy(RuleBasedPolicy):
```
Comment (lihuoran) [inline examples/supply_chain/rl/algorithms/rule_based.py:13]: Expected 2 blank lines, found 1

File: examples/supply_chain/rl/algorithms/rule_based.py [hunk 4]
```diff
@@ -1,127 +1,70 @@
...
+            expected_vlt * state["sale_mean"]
+            + math.sqrt(expected_vlt) * state["sale_std"] * state["service_level_ppf"]
+        )
+
+        capacity_mask = storage_booked_quantity <= state["storage_capacity"]
+        replenishment_mask = booked_quantity <= self._replenishment_threshold
+
+        return capacity_mask and replenishment_mask
+
+    def _get_action_quantity(self, state: dict) -> int:
```
Comment (lihuoran) [inline examples/supply_chain/rl/algorithms/rule_based.py:44]: Add `@abstractmethod`

File: examples/supply_chain/rl/policies.py [hunk 5]
```diff
@@ -51,19 +46,21 @@ def entity2policy(entity: SupplyChainEntity) -> str:
...
+    return None
 
 
 policy_creator = {
     "ppo.policy": partial(get_policy, STATE_DIM, NUM_CONSUMER_ACTIONS),
-    "manufacturer_policy": lambda name: ManufacturerBaselinePolicy(name),
-    "facility_policy": lambda name: DummyPolicy(name),
-    "product_policy": lambda name: DummyPolicy(name),
-    "seller_policy": lambda name: DummyPolicy(name),
+    "manufacturer_policy": lambda name: DummyPolicy(name),
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply examples/supply_chain/rl/policies.py:56]: yes, the action return by ManufactureBaselinePolicy is not used

File: maro/rl/rollout/env_sampler.py (hunk 1 above)
Comment (Jinyu-W) [reply maro/rl/rollout/env_sampler.py:286]: the action of a specific **agent** can be any object. Here the "state" is not the List of state for a **policy**

File: examples/supply_chain/rl/env_sampler.py (hunk 2 above)
Comment (Jinyu-W) [reply examples/supply_chain/rl/env_sampler.py:220]: None?

Comment (lihuoran) [reply examples/supply_chain/rl/env_sampler.py]: According to the return type, we should return a `np.ndarray` or a `dict`. If we want to return `None`, we could add `None` into the returned `Union`.

**Pull request merged**